Update replication example to use Ubuntu 22.04 and the latest PostgreSQL (#100)
pgroonga/pgroonga.github.io: reference/streaming-replication.md

  6. [special] [Add some PGroonga's WAL configurations to `postgresql.conf` on standbys](#special-add-some-pgroongas-wal-configurations-to-postgresqlconf-on-standbys)

  7. [normal] [Restart PostgreSQL on standbys](#normal-restart-postgresql-on-standbys)

  8. [normal] [Insert data on primary](#normal-insert-data-on-primary)

  9. [special] [Create a PGroonga index on primary](#special-create-a-pgroonga-index-on-primary)

## Example environment

This document uses the following environment:

  * Primary:

    * OS: Ubuntu 22.04

    * IP address: 192.168.0.30

    * Database name: `blog`

    * Replication user name: `replicator`

  * Standby1:

    * OS: Ubuntu 22.04

    * IP address: 192.168.0.31

  * Standby2:

    * OS: Ubuntu 22.04

    * IP address: 192.168.0.32

This document shows command lines for CentOS 7. If you're using other platforms, adjust command lines by yourself.

## [normal][special] Install PostgreSQL and PGroonga on both primary and standbys

Install PostgreSQL 15 on both primary and standbys.

Primary:

```bash
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -

> sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" >> /etc/apt/sources.list.d/pgdg.list'

Author: なるほど！勉強になります👍

sudo apt update
sudo apt install -y postgresql-15 postgresql-contrib-15

sudo apt install -y software-properties-common
sudo add-apt-repository -y universe
sudo add-apt-repository -y ppa:groonga/ppa
sudo apt install -y wget lsb-release
wget https://packages.groonga.org/ubuntu/groonga-apt-source-latest-$(lsb_release --codename --short).deb

> sudo apt install -y -V ./groonga-apt-source-latest-$(lsb_release --codename --short).deb

Author: @kou 

修正しました〜👍

Standbys:

> `/etc/postgresql/15/main/postgresql.conf`:

Reviewer: あぁ、Ubuntuのパッケージだと設定ファイルをデータディレクトリー以下におかずに`/etc/postgresql/`以下に置いているから`pg_basebackup`でコピーされないんですか。

PostgreSQLのバージョンが上がったから変わったわけじゃなくてUbuntuのパッケージの構成だとコピーされないだけなんですね。

Before:

```vim

> data_directory = '/var/lib/postgresql/15/main'
> 

Reviewer: ↑の`pg_basebackup`で既存のデータディレクトリーを上書きしませんか？

```suggestion
```

> #hot_standby = off

Reviewer: PostgreSQLだと`hot_standby = on`がデフォルトみたいなので変更する手順を削除してもらえますか？

Standbys:

`/etc/postgresql/15/main/postgresql.conf`:

Before:

```vim
#shared_preload_libraries = ''
```

After:

```vim

> shared_preload_libraries = 'pgroonga_wal_applier'

Reviewer: `pgroonga_wal_applier`はdeprecatedにしたんです。

ここの少し上にも`pgroonga_wal_applier`の話があるので更新してもらえますか？

```suggestion
shared_preload_libraries = 'pgroonga_standby_maintainer'
```

> # Probably add to the bottom of psotgresql.cong

Reviewer: ```suggestion
# Probably add to the bottom of postgresql.conf
```

> SELECT name,setting FROM pg_settings WHERE name = 'shared_preload_libraries';

Reviewer: このタイミングだとPostgreSQLが動いていないのでこれは確認できないですね。。。

消すか別の場所に移動するか。。。

Author: @kou 

消しました
m(_ _)m

```

> ## [normal] Restart PostgreSQL on standbys

Reviewer: `pg_basebackup`のときに止めたのでStartでいいです。あるいは、`pg_basebackup`直後に動かし直すかですねぇ。

```suggestion
## [normal] Start PostgreSQL on standbys
```

This is a normal step.

Restart PostgreSQL on standbys:

```bash
% sudo -H systemctl restart postgresql
```

## [normal] Insert data on primary

This is a normal step.

Create a database on only primary:

```bash

> sudo su - postgres
> 
> createdb blog

Reviewer: ```suggestion
sudo -u postgres -H createdb blog
```

```

Create a table in the created database on only primary.

Connect to the created `blog` database:

```bash

> sudo su - postgres
> 
> psql blog

Reviewer: ```suggestion
sudo -u postgres -H psql blog
```
